File: Medium/1980. Find Unique Binary String/FindUniqueBinaryString.java
/*
LeetCode Medium: 1980. Find Unique Binary String
https://leetcode.com/problems/find-unique-binary-string/

Check FindUniqueBinaryString.cpp for brief solution explanation.
*/
class Solution {
    public String findDifferentBinaryString(String[] nums) {
        int N = nums.length;
        String ans = "";
        for(int mask = 0;mask<(1<<N);mask++){
            String cur = "";
            for(int i = 0;i<N;i++){
                if((mask&(1<<i))>0){
                    cur = "1"+cur;
                }else{
                    cur = "0"+cur;
                }
            }
            boolean check = true;
            for(int i = 0;i<N;i++){
                if(cur.equals(nums[i])){
                    check = false;
                    break;
                }
            }
            if(check){
                ans = cur;
                break;
            }
        }
        return ans;
    }
}

File: Medium/1980. Find Unique Binary String/FindUniqueBinaryString.py
'''
LeetCode Medium: 1980. Find Unique Binary String
https://leetcode.com/problems/find-unique-binary-string/

Check FindUniqueBinaryString.cpp for brief solution explanation.
'''
class Solution:
    def findDifferentBinaryString(self, nums: List[str]) -> str:
        N = len(nums)
        ans = ""
        for mask in range(1<<N):
            cur = ""
            for i in range(N):
                if mask&(1<<i):
                    cur = "1"+cur
                else:
                    cur = "0"+cur
            check = True
            for i in range(N):
                if cur==nums[i]:
                    check = False
                    break
            if check:
                ans = cur
                break
        return ans